operations/ftdc.go

```diff
@@ -737,19 +737,86 @@ func toT2() cli.Command {
 			ctx, cancel := context.WithCancel(context.Background())
 			defer cancel()
 
-			inputFile, err := os.Open(inputPath)
+			inputStat, err := os.Stat(inputPath)
 			if err != nil {
 				return errors.Wrapf(err, "problem opening file '%s'", inputPath)
 			}
-			defer func() { grip.Warning(inputFile.Close()) }()
 
-			// All genny output files are named using the workload actor and operation.
-			//   i.e., Actor.Operation.ftdc
-			//
-			// We get the actor and operation names from the ftdc filepath for use
-			// in the translation process.
-			fileName := filepath.Base(inputPath)
-			actorOperation := strings.Split(fileName, ".ftdc")
+			// Stores ftdc chunk iterators and its metadata to be used in TranslateGenny.
+			var outputSlice []*ftdc.GennyOutputMetadata
+
+			// Check if input path is an individual file or directory.
+			// All ftdc files in the provided input directory are initially
+			// used to set GennyOutputMetadata Name, Iter, StartTime and EndTime.
+			switch mode := inputStat.Mode(); {
+			// Handle directory.
+			case mode.IsDir():
+				if !strings.HasSuffix(inputPath, "/") {
+					inputPath += "/"
+				}
+				input, err := os.Open(inputPath)
+				if err != nil {
+					return errors.Wrapf(err, "problem opening input path '%s'", inputPath)
+				}
+				defer func() { grip.Warning((input.Close())) }()
+
+				files, err := input.Readdir(-1)
+				if err != nil {
+					return errors.Wrapf(err, "problem reading dir '%s'", inputPath)
+				}
+
+				for _, file := range files {
+					if file.Mode().IsRegular() && filepath.Ext(file.Name()) == ".ftdc" {
+						var gennyOutput ftdc.GennyOutputMetadata
+						filePath := filepath.Join(inputPath + file.Name())
+
+						f, err := os.Open(filePath)
+						if err != nil {
+							return errors.Wrapf(err, "problem opening file '%s'", filePath)
+						}
+						defer func() { grip.Warning(f.Close()) }()
```

Comment:
Same this as below, instead of using a defer, close `f` once you no longer need it.